### PR TITLE
Implement Service Information Message.

### PR DESCRIPTION
**Pull request by argilo**
This adds SIS message type 6, Service Information Message, which encodes information about the available audio and data services. For my local station it prints the following:
```
21:40:30 DEBUG pids.c:291: Data service: public, type 65, MIME type 5b6
21:40:30 DEBUG pids.c:291: Data service: public, type 259, MIME type eb2
21:40:30 DEBUG pids.c:291: Data service: public, type 259, MIME type 469
21:40:30 DEBUG pids.c:291: Data service: public, type 66, MIME type 2d7
21:40:30 DEBUG pids.c:291: Data service: public, type 66, MIME type e96
21:40:30 DEBUG pids.c:269: Audio program 0: public, type 8, sound experience 0
21:40:30 DEBUG pids.c:269: Audio program 1: public, type 4, sound experience 0
21:40:30 DEBUG pids.c:269: Audio program 2: public, type 3, sound experience 0
```
For details see http://www.nrscstandards.org/sg/nrsc-5-d-reference-docs/1020s.pdf section 4.6.

This gets closer to completing #38.

### CODE REVIEW

**Comment by argilo** [comment]
I've also added SIS Parameter Message (type 7). From my local station, I get:
```
23:46:52 DEBUG pids.c:317: Pending leap second offset: 0, current leap second offset: 14
23:46:55 DEBUG pids.c:322: ALFN of pending leap second adjustment: 0
23:46:56 DEBUG pids.c:328: Local time zone offset: -300 minutes, DST sched. 6, local DST? yes, regional DST? yes
23:47:02 DEBUG pids.c:340: Exciter manuf. "GG", core version 4.4.7.0, manuf. version 1.9.0.1
23:47:05 DEBUG pids.c:353: Importer manuf. "  ", core version 4.4.7.0, manuf. version 0.0.0.0
```

**Comment by argilo** [comment]
Emergency Alerts Message (type 9) is now implemented as well. They aren't very common, but I found one in a recording I made of KXTE:

`10:58:44 DEBUG pids.c:444: Alert: The Civil Authorities have issued a Child Abduction Emergency for all of Nevada beginning at 9:09 pm Thu Jul  6 and ending at 12:39 am Tue Jul 11 (KXTE LP1)`

File: src/pids.c [hunk 1]
```diff
@@ -239,7 +242,60 @@ void decode_sis(pids_t *st, uint8_t *bits)
             }
             break;
         case 6:
-            off += 27;
+            if (off > 64 - 27) break;
+            category = decode_int(bits, &off, 2);
+            switch (category)
+            {
+            case 0:
```
Comment (tylert) [inline src/pids.c]: I would indent these here just to be consistent.

File: src/pids.c [hunk 2]
```diff
@@ -239,10 +243,122 @@ void decode_sis(pids_t *st, uint8_t *bits)
             }
             break;
         case 6:
-            off += 27;
+            if (off > 64 - 27) break;
```
Comment (tylert) [inline src/pids.c]: Should there be a log_debug here inside the if statement?

File: src/pids.c [hunk 3]
```diff
@@ -239,10 +243,122 @@ void decode_sis(pids_t *st, uint8_t *bits)
...
+                        log_debug("Importer manuf. \"%c%c\", core version %d.%d.%d.%d-%d, manuf. version %d.%d.%d.%d-%d",
+                            (st->parameters[8] >> 8) & 0x7f, st->parameters[8] & 0x7f,
+                            (st->parameters[9] >> 11) & 0x1f, (st->parameters[9] >> 6) & 0x1f, (st->parameters[9] >> 1) & 0x1f,
+                            (st->parameters[11] >> 11) & 0x1f, (st->parameters[11] >> 3) & 0x7,
+                            (st->parameters[10] >> 11) & 0x1f, (st->parameters[10] >> 6) & 0x1f, (st->parameters[10] >> 1) & 0x1f,
+                            (st->parameters[11] >> 6) & 0x1f, st->parameters[11] & 0x7
+                        );
+                    }
+                    break;
+                }
```
Comment (tylert) [inline src/pids.c]: Maybe add a default here with a log_debug message?

File: src/pids.h [hunk 4]
```diff
@@ -26,11 +44,24 @@ typedef struct
...
+
+    int parameters[NUM_PARAMETERS];
+
     char slogan[96];
     uint8_t slogan_have_frame[16];
     int slogan_encoding;
     int slogan_len;
     int slogan_displayed;
+
+    char alert[382];
```
Comment (tylert) [inline src/pids.h]: Make defines for these fixed values too just like MAX_DATA_SERVICES?  Nice for improving readability.

**Comment by tylert** [comment]
@argilo LGTM.  🚢 IT!

File: src/pids.c (hunk 3 above)
Comment (argilo) [reply src/pids.c]: The `index` is already checked above, so no other value is possible.

File: src/pids.h (hunk 4 above)
Comment (argilo) [reply src/pids.h]: Good idea. I'll do the same for the other magic numbers in pids.c and pids.h.

File: src/pids.c (hunk 3 above)
Comment (tylert) [reply src/pids.c]: Awesome.  I usually add a comment whenever I deliberately skip a default case in a switch statement.  Cool.

**Pull request merged**